src/bp/core/repositories/telemetry_payload.ts

```diff
@@ -0,0 +1,53 @@
+import { DataRetentionService } from 'core/services/retention/service'
+import { inject, injectable } from 'inversify'
+import Knex from 'knex'
+import _ from 'lodash'
+import ms from 'ms'
+
+import Database from '../database'
+import { TYPES } from '../types'
+
+export interface TelemetryPayloadRepository {
+  getPayload(uuid: string): Promise<any>
+  insertPayload(uuid: string, url: string, payload: JSON): Promise<void>
+  removePayload(uuid: string): Promise<any>
+  getAll(): Promise<any>
+  getN(n: number): Promise<any>
+}
+
+@injectable()
+export class KnexTelemetryPayloadRepository implements TelemetryPayloadRepository {
+  private readonly tableName = 'telemetry_table'
+
+  constructor(@inject(TYPES.Database) private database: Database) {}
+
+  async getN(n: number): Promise<any> {
+    return await this.database.knex
+      .from(this.tableName)
+      .select('*')
+      .limit(n)
+  }
+
+  async getAll(): Promise<any> {
+    return await this.database.knex.from(this.tableName).select('*')
+  }
+
+  async insertPayload(uuid: string, url: string, payload: JSON) {
+    await this.database.knex(this.tableName).insert({ uuid: uuid, url: url, payload: JSON.stringify(payload) })
+  }
+
+  async getPayload(uuid: string): Promise<any> {
+    const payload = await this.database.knex
+      .from(this.tableName)
+      .select('*')
+      .where('uuid', uuid)
+    return payload
+  }
+
+  async removePayload(uuid: string): Promise<any> {
+    await this.database
+      .knex(this.tableName)
+      .where({ uuid: uuid })
```

Comment:
You can use the shorthand syntax:

```suggestion
      .where({ uuid })
```